web/client-api/src/main/java/io/deephaven/web/client/api/CoreClient.java

```diff
@@ -54,7 +81,12 @@ public Promise<Void> relogin(String token) {
     }
 
     public Promise<String[][]> getServerConfigValues() {
-        return Promise.resolve(new String[0][]);
+        return getConfigs(
```

Comment:
Just making sure - these are general server information, and we don't expect the user to be authenticated before retrieving them.

Reply:
Ah good catch - the server will have to auth this and fail the user in this case. i'll double check what dhe does here, but I think this is the same?

Reply:
Yes, on DHE it checks if you're authenticated for server config values: https://gitlab.deephaven.io/illumon/iris/-/blob/rc/silverheels/web/server-frontend/src/main/java/com/illumon/iris/web/server/WebApiServerImpl.java#L844
You can get the authConfigValues before being authenticated though (as you need them to know what you can login with): https://gitlab.deephaven.io/illumon/iris/-/blob/rc/silverheels/web/server-frontend/src/main/java/com/illumon/iris/web/server/WebApiServerImpl.java#L384
Also looks like there's a minor bug in DHE - if you try `client.getServerConfigValues()` before logged in, you'll get a `Not logged in` error (which makes sense), however if you then login with that same `client` object and call `client.getServerConfigValues()` afterward, you still get a `Not logged in` error. This is because the serverConfigValues is just a lazy that initializes the first time the get is called, and does not reset even if there was an error (or you login): https://gitlab.deephaven.io/illumon/iris/-/blob/rc/silverheels/web/client-api/src/main/java/com/illumon/iris/web/client/api/IrisClient.java#L80
Such a minor issue.

Reply:
I'm leaving this as is, re-querying the server each time, to avoid the bug you describe. The tradeoff is that if the client asks for the config again, they will get an extra round trip, and in theory could get different values, going by the current implementation. That might be a feature too, letting the client poll for values.